tools/deployment/k8s/s3/deploy.sh

```
@@ -0,0 +1,149 @@
+create_and_configure_bucket() {
+  # Create log-viewer bucket if not already exist
+  log "INFO" "Creating s3://${LOG_VIEWER_BUCKET} bucket."
+  if ! aws s3api head-bucket --endpoint-url "$AWS_ENDPOINT_URL" --bucket "$LOG_VIEWER_BUCKET" 2>/dev/null; then
+      aws s3api create-bucket --endpoint-url "$AWS_ENDPOINT_URL" --bucket "$LOG_VIEWER_BUCKET"
+  else
+      log "WARN" "Bucket s3://${LOG_VIEWER_BUCKET} already exist."
+  fi
+
+  # Define and apply the Bucket Policy for public read access
+  log "INFO" "Applying public read access policy to s3://${LOG_VIEWER_BUCKET}"
+  readonly POLICY=$(
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)